frontend/app/src/main/java/com/example/speechbuddy/viewmodel/SignupViewModel.kt

```diff
@@ -60,17 +63,26 @@ class SignupViewModel @Inject internal constructor(
     }
 
     private fun validateNickname() {
-        if (nicknameInput.isNotEmpty() && nicknameInput.length <= Constants.MAXIMUM_NICKNAME_LENGTH) {
+        if (isValidNickname(nicknameInput)) {
             _uiState.update { currentSate ->
                 currentSate.copy(
```

Comment:
```suggestion
            _uiState.update { currentState ->
                currentState.copy(
```
넘 사소하지만 갑자기 타이포가 보여서,,!!!